drivers/sensor/tdk/icm4268x/CMakeLists.txt

```diff
@@ -0,0 +1,16 @@
+# SPDX-License-Identifier: Apache-2.0
+
+zephyr_library()
+
+zephyr_library_sources(
+    icm4268x.c
+    icm4268x_common.c
+    icm4268x_spi.c
+)
+
+zephyr_library_sources_ifdef(CONFIG_SENSOR_ASYNC_API icm4268x_rtio.c)
+zephyr_library_sources_ifdef(CONFIG_ICM4268X_DECODER icm4268x_decoder.c)
+zephyr_library_sources_ifdef(CONFIG_ICM4268X_STREAM icm4268x_rtio_stream.c)
+zephyr_library_sources_ifdef(CONFIG_ICM4268X_TRIGGER icm4268x_trigger.c)
```

Comment:
Of all the APIs the Sensor API has currently, I believe the trigger API is the least usable among them and conflicts the most with the stream API. I'm wondering if perhaps this API is one we really should aim to deprecate.

Supporting the polling read and stream is fairly easy I think. Supporting trigger and stream seems conflict ridden.